includes/Upgrade/Upgrades/V_3_3_1.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace WeDevs\Dokan\Upgrade\Upgrades;
+
+use WeDevs\Dokan\Abstracts\DokanUpgrader;
+
+class V_3_3_1 extends DokanUpgrader {
+
+    /**
+     * Updates withdraw database table column
+     *
+     * @since DOKAN_LITE_SINCE
+     *
+     * @return void
+     */
+    public static function update_withdraw_table_column() {
+        global $wpdb;
+
+        include_once ABSPATH . 'wp-admin/includes/upgrade.php';
+
+        $existing_columns = $wpdb->get_col( "DESC `{$wpdb->prefix}dokan_withdraw`", 0 );
+
+        if ( in_array( 'details', $existing_columns, true ) ) {
+            return;
+        }
+
+        $wpdb->query(
+            "ALTER TABLE `{$wpdb->prefix}dokan_withdraw` ADD COLUMN `details` longtext AFTER `note`" // phpcs:ignore
+        );
+    }
+
+    /**
+     * Updates withdraw database table column
+     *
+     * @since DOKAN_LITE_SINCE
+     *
+     * @return void
+     */
+    public static function update_refund_table_column() {
+        global $wpdb;
+
+        include_once ABSPATH . 'wp-admin/includes/upgrade.php';
+
+        $existing_columns = $wpdb->get_results( "DESC `{$wpdb->prefix}dokan_refund`" );
+
+        if ( ! empty( $existing_columns ) ) {
```

Comment:
You don't need to check if it is empty. Just cast the `$existing_columns` by `array` while running foreach loop. As the `$wpdb::get_results` method return `array|object|null`, if the value is null, the array casting will convert it into an empty array and for empty array the foreach loop will not run automatically.
Rewrite this in the following way:

```
    foreach ( (array) $existing_columns as $existing_column ) {
        if (
            'text' !== $existing_column->Type &&
            (
                'item_totals' === $existing_column->Field ||
                'item_tax_totals' === $existing_column->Field
            )
        ) {
            $wpdb->query(
                "ALTER TABLE `{$wpdb->prefix}dokan_refund` MODIFY COLUMN {$existing_column->Field} text" // phpcs:ignore
            );
        }
    }
```